core/node_path.cpp

```diff
@@ -407,7 +407,7 @@ NodePath::NodePath(const String &p_path) {
 
 	data = memnew(Data);
 	data->refcount.init();
-	data->absolute = absolute ? true : false;
+	data->absolute = absolute != 0;
```

Comment:
I'd make `absolute` a bool above and do an explicit `(int)absolute` cast in the `for` loop.